src/main/kotlin/lotto/LottoMachine.kt

```diff
@@ -1,42 +1,17 @@
 package lotto
 
-import kotlin.math.floor
-
-class LottoMachine(private val amount: Int) {
-
-    fun getCountOfLotto(): Int {
-        require(this.amount >= 0 && this.amount >= LOTTO_PRICE) { "$LOTTO_PRICE 원 이상으로 금액을 입력해주세요." }
-        return (this.amount / LOTTO_PRICE)
-    }
-
-    fun getLottoList(): List<Lotto> {
-        return (1..getCountOfLotto()).map { Lotto() }
-    }
-
-    companion object {
-        private const val LOTTO_PRICE = 1_000
-
-        fun checkWinningNumbers(lottos: List<Lotto>, winningNumbers: List<Int>): Map<Int, Int> {
-            val winningCountMap = mutableMapOf<Int, Int>()
-            for (lotto in lottos) {
-                val matchingNumbers = lotto.generateCachedLottoNumbers().count { it in winningNumbers }
-                winningCountMap[matchingNumbers] = winningCountMap.getOrDefault(matchingNumbers, 0) + 1
-            }
-            return winningCountMap
+object LottoMachine {
+    private val LOTTO_PRICE = 1000
+    fun generateLotto(amount: Int, lottoNumberGenerator: LottoNumberGenerator): List<Lotto> {
+        val lottoCount = amount / LOTTO_PRICE
+        val lottoList = mutableListOf<Lotto>()
+
+        repeat(lottoCount) {
+            val lotto = Lotto.from(lottoNumberGenerator)
+            lottoList.add(lotto)
```

Comment:
저는 개인적으로 mutableXXX 생성을 최대한 지양하는 편이에요. 로직 사이에 어떤 위변조에 취약해질 수 있다고 생각하거든요. 좀 더 immutable하게 만들수도 있을 것 같네요
```suggestion
        return (0 until lottoCount).map { Lotto.from(generator) }
```

Reply:
오 네넵!! 훨씬 간결해지네요!